src/main/java/com/conveyal/r5/profile/FastRaptorWorker.java

```diff
@@ -323,74 +348,61 @@ private void advanceScheduledSearchToPreviousMinute (int nextMinuteDepartureTime
                 doScheduledSearchForRound(scheduleState[round - 1], scheduleState[round]);
                 raptorTimer.scheduledSearchTransit.stop();
 
-                // If there are frequency routes, we will be randomizing the offsets of those routes.
+                // If there are frequency routes, we will be randomizing the schedules of those routes.
                 // First perform a frequency search using worst-case boarding time to provide a tighter upper bound on
-                // total travel time. Each randomized schedule will improve on these travel times.
-                // This is only helpful for Monte Carlo, not half-headway. Even in Monte Carlo mode,
-                // perhaps we should only do it when iterationsPerMinute is high (2 or more?).
-                if (transit.hasFrequencies && boardingMode == MONTE_CARLO) {
+                // total travel time. Each randomized schedule will then improve on these travel times.
+                // This should be a pure optimization, which is only helpful for Monte Carlo mode (not half-headway).
+                // Perhaps we should only do it when iterationsPerMinute is high (2 or more?)
+                if (transit.hasFrequencies && boardingMode == MONTE_CARLO && ENABLE_OPTIMIZATIONS) {
                     raptorTimer.scheduledSearchFrequencyBounds.start();
                     doFrequencySearchForRound(scheduleState[round - 1], scheduleState[round], UPPER_BOUND);
                     raptorTimer.scheduledSearchFrequencyBounds.stop();
                 }
-
+                // Apply transfers to the scheduled result that will be reused for the previous departure minute.
+                // Transfers will be applied separately to the derived frequency result below, when relevant.
                 raptorTimer.scheduledSearchTransfers.start();
                 doTransfers(scheduleState[round]);
                 raptorTimer.scheduledSearchTransfers.stop();
             }
             raptorTimer.scheduledSearch.stop();
         }
 
-        // Now run frequency searches using randomized schedules for all frequency lines. We use the scheduled search
-        // and the worst-case boarding wait for all frequency routes to establish an upper bound on arrival times for
-        // the frequency searches, so we are copying the arrival times from the just completed search.
+        // Now run one or more frequency searches using randomized schedules for all frequency lines.
+        // These will improve upon any upper bound travel times just established at this departure minute.
         // This is a key innovation, described in more detail in Conway, Byrd, and van der Linden 2017.
+        // Unlike those scheduled results, these frequency results are not reusable at the previous minute because
+        // the schedule is almost certain to be different at that previous minute.
```

Comment:
As explained in the 2017 paper,
> we randomize the schedules at each departure minute of a single range-RAPTOR search. This means that the range-RAPTOR approach of bounding each search with the results of a search at a later minute is no longer applicable

The range-RAPTOR optimization (using arrival times for departure time _t_ as an upper bound on arrival times for departure time _t_ - 1) works for the patterns with true schedules or worst-case (full headway) waiting times, but not the patterns with randomized schedules.